eng/bomgenerator/src/main/java/com/azure/tools/bomgenerator/Utils.java

```diff
@@ -89,18 +114,33 @@ else if (groups.length == 3) {
             return HTTP_CLIENT.sendAsync(request, HttpResponse.BodyHandlers.ofInputStream())
                 .thenApply(response -> {
                     if(response.statusCode() == 200) {
-                        return Utils.parsePomFileContent(response.body());
+                        try (InputStreamReader reader = new InputStreamReader(response.body())) {
+                            return Utils.parsePomFileContent(reader);
+                        }
+                        catch (IOException ex) {
+                            logger.error("Failed to read contents for {}", dependency.toString());
```

Comment:
If we cannot read the response, should we just abort the BOM generation? Can we proceed to generate a valid BOM without this dependency?

Reply:
We can generate the right BOM - we may just not be able to recognize that an external dependency is getting resolved by an external BOM we have added. 

The tool, takes the stance of generating the report with/without errors. So, I do not stop the analysis in case of any errors.